### PR TITLE
xml annotations support (bug #82)

### PR DESCRIPTION
**Pull request by emmanueltouzery**
fix bug #82. Actually after your commit it was already working, it was just a flaw in the test itself :-)

I added a Set test, you mentioned adding a Seq test but I'm not sure because Seq is abstract? Let me know what you would add and I can add it!

PS: I guess we'll have to wait for the next javaslang release for this to be released after being merged?

### CODE REVIEW

File: src/test/java/javaslang/jackson/datatype/seq/ListTest.java [hunk 1]
```diff
@@ -66,4 +74,55 @@ public void testJsonTypeInfo() throws IOException {
...
+            transitTypes.add(1);
+            transitTypes.add(2);
+            transitTypes.add(3);
+        }
+    }
+
+    @Test
+    public void testXmlSerialization() throws IOException {
+        String javaUtilValue = xmlMapper().writeValueAsString(new XmlSerializeJavaslang());
+        Assert.assertEquals(xmlMapper().writeValueAsString(new XmlSerializeJavaUtil()), javaUtilValue);
```
Comment (ruslansennov) [inline src/test/java/javaslang/jackson/datatype/seq/ListTest.java]: Something wrong with deserialization of XML-source. Here I added these lines:
```java
XmlSerializeJavaslang restored = xmlMapper().readValue(javaUtilValue, XmlSerializeJavaslang.class);
Assert.assertEquals(restored.transitTypes.size(), 3);
```
Process fails with error:
```
com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.lang.Integer out of FIELD_NAME token
 at [Source: java.io.StringReader@53aad5d5; line: 1, column: 51] (through reference chain: javaslang.jackson.datatype.seq.XmlSerializeJavaslang["transitTypes"])

	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:148)
	at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:857)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseInteger(StdDeserializer.java:460)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:309)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:287)
	at javaslang.jackson.datatype.deserialize.ArrayDeserializer.deserialize(ArrayDeserializer.java:52)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:101)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:258)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3736)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2726)
	at javaslang.jackson.datatype.seq.ListTest.testXmlSerialization(ListTest.java:129)
       .....
```

Comment (ruslansennov) [inline src/test/java/javaslang/jackson/datatype/seq/ListTest.java]: `java.util.List` decoded fine:
```java
        XmlSerializeJavaUtil restored = xmlMapper().readValue(javaUtilValue, XmlSerializeJavaUtil.class);
        Assert.assertEquals(restored.transitTypes.size(), 3);
```

**Comment by ruslansennov** [comment]
> PS: I guess we'll have to wait for the next javaslang release for this to be released after being merged?

@danieldietrich [says](https://github.com/javaslang/javaslang/pull/1646#discussion_r85347725) today that Javaslang version `2.1.0` will not be ready soon. So maybe I'll prepare another bugfix release based on Javaslang `2.0.4`

**Comment by danieldietrich** [comment]
We will release a 2.1.0-beta next month and then go through the RC phase in able to ship the most stable version possible. There are quite a few new features and collections - these need to be extensively tested.

File: src/test/java/javaslang/jackson/datatype/seq/ListTest.java (hunk 1 above)
Comment (ruslansennov) [inline src/test/java/javaslang/jackson/datatype/seq/ListTest.java]: It seems this is not XML-related problem. I'll try to fix it in own PR.

**Pull request merged**